### PR TITLE
refactor(app): show additional required labware during setup

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

Include addition "off deck" labware in the labware setup instructions within the protocol run setup flow.

In addition to the specifying the locations of the labware that start on the deck of the robot
during protocol setup, include descriptions of the labware that an operator must have on hand to
complete the protocol,that doesn't begin in a deck location.

Closes [RLAB-155](https://opentrons.atlassian.net/browse/RLAB-155)

# Changelog

- modernize the design to the Run Labware Position Check CTA
- show additional labware if present underneath the map view and list view 
- if there are applied offsets on a run, show a "view current offsets" link that launches a table showing all of the current offsets
- TODO for a follow-up add support for the jupyter/cli python snippet with DSM offsets 

# Risk assessment
low

### CODE REVIEW

File: app/src/organisms/Devices/ProtocolRun/SetupLabware/LabwareListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+} from '@opentrons/shared-data'
+import { useCreateLiveCommandMutation } from '@opentrons/react-api-client'
+import { ToggleButton } from '../../../../atoms/buttons'
+import { StyledText } from '../../../../atoms/text'
+import { SecureLabwareModal } from '../../../ProtocolSetup/RunSetupCard/LabwareSetup/SecureLabwareModal'
+import type {
+  HeaterShakerCloseLatchCreateCommand,
+  HeaterShakerOpenLatchCreateCommand,
+} from '@opentrons/shared-data/protocol/types/schemaV6/command/module'
+import type { ModuleTypesThatRequiresExtraAttention } from '../../../ProtocolSetup/RunSetupCard/LabwareSetup/utils/getModuleTypesThatRequireExtraAttention'
```
Comment (smb2268) [inline app/src/organisms/Devices/ProtocolRun/SetupLabware/LabwareListItem.tsx]: nit / can be fixed later - but `Require` here shouldn't be plural

File: app/src/organisms/Devices/ProtocolRun/SetupLabware/LaunchLabwarePositionCheck.tsx [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+  }
+
+  return (
+    <Flex
+      gridGap={SPACING.spacing4}
+      backgroundColor={COLORS.lightBlue}
+      padding={SPACING.spacing4}
+      borderRadius={BORDERS.radiusSoftCorners}
+    >
+      <Icon name="reticle" size="18px" />
```
Comment (jerader) [inline app/src/organisms/Devices/ProtocolRun/SetupLabware/LaunchLabwarePositionCheck.tsx:128]: oooh i didn't know what a `reticle` was until now. pretty neat

**Review by jerader — COMMENTED** [review]
looking pretty good! 🚀 

One thing I noticed: if you have too many additional labware, LPC banner gets cut off:

Screen Shot 2022-11-03 at 3 33 53 PM

**Review by smb2268 — APPROVED** [review]
Let's merge 💪🏻

File: app/src/organisms/Devices/ProtocolRun/SetupLabware/CurrentOffsetsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  Link,
+} from '@opentrons/components'
+import { getIsLabwareOffsetCodeSnippetsOn } from '../../../../redux/config'
+import { ModalHeader, ModalShell } from '../../../../molecules/Modal'
+import { OffsetVector } from '../../../../molecules/OffsetVector'
+import { getDefsByURI } from '../../../ApplyHistoricOffsets/hooks/getDefsByURI'
+
+import type { LabwareOffset } from '@opentrons/api-client'
+import { PrimaryButton } from '../../../../atoms/buttons'
+import { useSelector } from 'react-redux'
```
Comment (koji) [inline app/src/organisms/Devices/ProtocolRun/SetupLabware/CurrentOffsetsModal.tsx]: can we move up these lines?

File: app/src/organisms/Devices/ProtocolRun/SetupLabware/CurrentOffsetsModal.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+          width="100%"
+          marginTop={SPACING.spacing6}
+          justifyContent={JUSTIFY_FLEX_END}
+          alignItems={ALIGN_CENTER}
+          gridGap={SPACING.spacing3}
+        >
+          <Link
+            color={COLORS.blueEnabled}
+            fontSize={TYPOGRAPHY.fontSizeP}
+            fontWeight={TYPOGRAPHY.fontWeightSemiBold}
```
Comment (koji) [inline app/src/organisms/Devices/ProtocolRun/SetupLabware/CurrentOffsetsModal.tsx]:
```suggestion
css={TYPOGRAPHY.linkPSemiBold}
```

File: app/src/organisms/Devices/ProtocolRun/SetupLabware/LaunchLabwarePositionCheck.tsx [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+        alignItems={ALIGN_CENTER}
+        justifyContent={JUSTIFY_FLEX_END}
+        flex="1 0 auto"
+        gridGap={SPACING.spacing4}
+      >
+        {currentOffsets.length > 0 ? (
+          <Link
+            color={COLORS.blueEnabled}
+            fontSize={TYPOGRAPHY.fontSizeP}
+            fontWeight={TYPOGRAPHY.fontWeightSemiBold}
```
Comment (koji) [inline app/src/organisms/Devices/ProtocolRun/SetupLabware/LaunchLabwarePositionCheck.tsx]: same as the above

File: app/src/organisms/Devices/ProtocolRun/SetupLabware/OffDeckLabwareList.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import * as React from 'react'
+import { SPACING, TYPOGRAPHY } from '@opentrons/components'
+import { useTranslation } from 'react-i18next'
```
Comment (koji) [inline app/src/organisms/Devices/ProtocolRun/SetupLabware/OffDeckLabwareList.tsx]:
```suggestion
import { useTranslation } from 'react-i18next'
import { SPACING, TYPOGRAPHY } from '@opentrons/components'
```

**Pull request merged**